activesupport/lib/active_support/railtie.rb

```diff
@@ -27,6 +27,12 @@ class Railtie < Rails::Railtie # :nodoc:
       end
     end
 
+    initializer "active_support.reset_execution_context" do |app|
+      app.reloader.before_class_unload { ActiveSupport::ExecutionContext.clear }
+      app.executor.to_run              { ActiveSupport::ExecutionContext.clear }
+      app.executor.to_complete         { ActiveSupport::ExecutionContext.clear }
```

Comment:
I wish we had a single API to set up all these 3 callbacks, seem to remember we're doing this in a bunch of places. Though for another time.

Reply:
Definitely.